include/evmc/evmc.hpp

```diff
@@ -27,6 +29,33 @@ class result : private evmc_result
     using evmc_result::output_size;
     using evmc_result::status_code;
 
+    /// Creates the result from the provided arguments.
+    ///
+    /// The provided output is copied to memory allocated with malloc()
+    /// and the evmc_result::release function is set to one invoking free().
+    ///
+    /// @param _status_code  The status code.
+    /// @param _gas_left     The amount of gas left.
+    /// @param _output_data  The pointer to the output.
+    /// @param _output_size  The output size.
+    result(evmc_status_code _status_code,
+           int64_t _gas_left,
+           const uint8_t* _output_data,
+           size_t _output_size) noexcept
+      : evmc_result{_status_code, _gas_left, nullptr, _output_size, {}, {}, {}}
+    {
+        if (output_size != 0)
+        {
+            auto mem = static_cast<uint8_t*>(std::malloc(output_size));
```

Comment:
Apparently this can return `nullptr`. Why not use `new` and `delete` here instead?

Reply:
Not practical. I want to keep VM implementation low profile and require only C standard library if possible.

But good point raised. How about converting it to `EVMC_INTERNAL_ERROR`?

Reply:
> How about converting it to EVMC_INTERNAL_ERROR?

I think that's a bad idea, because it masks what's going on. The caller would think the called contract failed.

Reply:
I'm not sure what do you mean by a caller here but in any case it should not take EVMC internal errors as indicators for contract execution failures.

I believe we only have 2 options here: ignore it or report it as an internal error.